modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/shared/ClayToggle.es.js

```diff
@@ -0,0 +1,62 @@
+import React, {Component} from 'react';
+import {PropTypes} from 'prop-types';
+import ClayIcon from './ClayIcon.es';
+
+class ClayToggle extends Component {
+	static propTypes = {
+		checked: PropTypes.bool,
+		className: PropTypes.string,
+		iconOff: PropTypes.string,
+		iconOn: PropTypes.string,
+		label: PropTypes.string,
+		onChange: PropTypes.func
+	};
+
+	static defaultProps = {
+		label: ''
+	};
+
+	render() {
+		const {
+			checked,
+			iconOff,
+			iconOn,
+			label,
+			onChange,
+			...otherProps
+		} = this.props;
+
+		return (
+			<label className="toggle-switch" {...otherProps}>
```

Comment:
Careful with adding `{...otherProps}` **after** your props. If the consumer added a className prop, it would overwrite your className.

Reply:
I'll extract out the `className` from the prop in the same way `ClaySelect` does.
```
		const classes = getCN(
			'toggle-switch',
			className
		);
```